### PR TITLE
Added code example for the stream operator map

### PR DESCRIPTION
**Pull request by gtlcheung**

### CODE REVIEW

File: akka-docs/src/test/scala/docs/stream/operators/Map.scala [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import akka.stream.scaladsl._
+
+//#imports
+
+object Map {
+
+  //#map
+  def mapExample: Source[String, NotUsed] = {
+    val source = Source(1 to 10)
+    source.map(elem => elem.toString)
```
Comment (raboof) [inline akka-docs/src/test/scala/docs/stream/operators/Map.scala]: Might be nice to add the types, to show that this `map` changes the `Source[Int]` into a `Source[String]`?

**Review by raboof — COMMENTED** [review]
Thanks for starting on this example! Might be nice to add some types to make it more obvious the types are checked?

**Comment by raboof** [comment]
OK TO TEST

File: akka-docs/src/test/scala/docs/stream/operators/Map.scala (hunk 1 above)
Comment (gtlcheung) [reply akka-docs/src/test/scala/docs/stream/operators/Map.scala]: Added the type to the source before applying the map.

File: akka-docs/src/test/scala/docs/stream/operators/Map.scala [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import akka.stream.scaladsl._
+
+//#imports
+
+object Map {
+
+  //#map
+  def mapExample: Source[String, NotUsed] = {
+    val source: Source[Int, NotUsed] = Source(1 to 10)
+    source.map(elem => elem.toString)
```
Comment (raboof) [inline akka-docs/src/test/scala/docs/stream/operators/Map.scala]: Wrapping it in a function might be confusing to new users... perhaps we could assign it like `val mapped: Source[String, NotUsed] = source.map(elem => elem.toString)`?

Comment (gtlcheung) [reply akka-docs/src/test/scala/docs/stream/operators/Map.scala]: You are right. Changed the example.

**Review by raboof — APPROVED** [review]
Nice!

**Review by chbatey — APPROVED** [review]
LGTM thanks!

**Pull request merged**